runtime/lua/vim/lsp/util.lua

```diff
@@ -1774,37 +1774,55 @@ end
 ---
 ---@param symbols lsp.DocumentSymbol[]|lsp.SymbolInformation[]
 ---@param bufnr? integer
+---@param position_encoding 'utf-8'|'utf-16'|'utf-32'
 ---@return vim.quickfix.entry[] # See |setqflist()| for the format
-function M.symbols_to_items(symbols, bufnr)
+function M.symbols_to_items(symbols, bufnr, position_encoding)
   bufnr = bufnr or 0
```

Comment:
I would use the newly added `vim._resolve_bufnr` here: https://github.com/neovim/neovim/pull/31499